src/mod_muc_room.erl

```diff
@@ -3211,32 +3212,45 @@ send_kickban_presence1(UJID, Reason, Code, Affiliation, StateData) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 % Owner stuff
 
--spec process_iq_owner(jid:jid(), get | set, ejabberd:lang(), exml:element(), state()) ->
+-spec process_iq_owner(jid:jid(), get | set, ejabberd:lang(), exml:element(),
+                       state(), statename()) ->
     {error, exml:element()} | {result, [exml:element() | jlib:xmlcdata()], state() | stop}.
-process_iq_owner(From, Type, Lang, SubEl, StateData) ->
+process_iq_owner(From, Type, Lang, SubEl, StateData, StateName) ->
     case get_affiliation(From, StateData) of
         owner ->
-            process_authorized_iq_owner(From, Type, Lang, SubEl, StateData);
+            process_authorized_iq_owner(From, Type, Lang, SubEl, StateData, StateName);
         _ ->
             ErrText = <<"Owner privileges required">>,
             {error, mongoose_xmpp_errors:forbidden(Lang, ErrText)}
     end.
 
--spec process_authorized_iq_owner(jid:jid(), get | set, ejabberd:lang(),
-                                  exml:element(), state()) ->
+-spec process_authorized_iq_owner(jid:jid(), get | set, ejabberd:lang(), exml:element(),
+                                  state(), statename()) ->
     {error, exml:element()} | {result, [exml:element() | jlib:xmlcdata()], state() | stop}.
-process_authorized_iq_owner(From, set, Lang, SubEl, StateData) ->
+process_authorized_iq_owner(From, set, Lang, SubEl, StateData, StateName) ->
     #xmlel{children = Els} = SubEl,
     case xml:remove_cdata(Els) of
         [#xmlel{name = <<"x">>} = XEl] ->
             case {xml:get_tag_attr_s(<<"xmlns">>, XEl),
-                  xml:get_tag_attr_s(<<"type">>, XEl)} of
-                {?NS_XDATA, <<"cancel">>} ->
+                  xml:get_tag_attr_s(<<"type">>, XEl),
+                  StateName,
+                  StateData#state.configuration_initiated} of
```

Comment:
ok, so if we WANT to keep it.
there can be more than one room administrator.

configuration_initiated should be PER room administrator device jid (i.e. alice@localhost/res1).
we also should reset configuration_initiated, when the administrator leaves the room (it's just logical, not something described in XEP).

Than, extra cases that should be tested and documented.

Admin asks for the form several times.
Admin sends the same form several times.
Admin sends the form once. Then sends the form again with different data.
There are two admins, one initiates configuration. one sends the form (AND VARIATIONS).
There are two admins, both fetched form and both submitted (last write wins? or do we want to check that the form is not expired yet?).

All these cases should be documented.
It's actually the main reason, why there was no check for the room.
Because any extra "if" should be tested.

Error reporting should be consistent and contain description, in which there should be information, that allows to find corresponding XEP part AND corresponding code in MongooseIM.

The problem that the error has no example in the XEP, so we should say something like "Form MUST be fetched before being submitted. See xep-0045, version 1.31.1. 10.1 Creating a Room => 10.1.1 General Considerations => workflow step 3.".

We need that step, because not all front-end developers read XEP, they just use libraries, which can have the form fetching step missing.